hawkbit-ddi-resource/src/main/java/org/eclipse/hawkbit/ddi/rest/resource/DdiArtifactStoreController.java

```diff
@@ -134,10 +134,10 @@
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
-    private ActionStatus checkAndReportDownloadByTarget(final HttpServletRequest request, final String targetid,
+    private ActionStatus checkAndReportDownloadByTarget(final HttpServletRequest request, final String controllerId,
             final Artifact artifact) {
-        final Target target = controllerManagement.updateLastTargetQuery(targetid,
-                IpUtil.getClientIpFromRequest(request, securityProperties));
+        final Target target = controllerManagement.findByControllerId(controllerId)
+                .orElseThrow(() -> new EntityNotFoundException(Target.class, controllerId));
```

Comment:
The field `securityProperties` in line 66 is now not used anymore, so you can remove it